llpc/context/llpcCompiler.cpp

```diff
@@ -1235,13 +1234,15 @@ Result Compiler::buildPipelineInternal(Context *context, ArrayRef<const Pipeline
           timerProfiler.getTimer(TimerCodeGen),
       };
 
-      pipeline->generate(std::move(pipelineModule), elfStream, checkShaderCacheFunc, timers, cl::NewPassManager == 2);
-#if LLPC_ENABLE_EXCEPTION
-      result = Result::Success;
-#endif
+      if (pipeline->generate(std::move(pipelineModule), elfStream, checkShaderCacheFunc, timers,
+                             cl::NewPassManager == 2))
+        result = Result::Success;
+      else
+        result = Result::ErrorInvalidShader;
```

Comment:
I don't think this is right. generate() returning false is for a _recoverable_ error where the front-end might want to retry but this time doing a whole pipeline compile.